src/test/kotlin/race/RandomTest.kt

```diff
@@ -8,7 +8,7 @@ class RandomTest {
     @Test
     fun `항상 0-9사이의 값이 나온다`() {
         // Arrange
-        val randomGeneratorNumber = RandomGeneratorNumber()
+        val randomGeneratorNumber = RandomNumberGenerator()
```

Comment:
강의에서 배운 내용을 상기해보면서, NumberGenerator를 익명 클래스로 생성해서 테스트해보는 것은 어떨까요?
Random 자체를 테스트하는 것은 무의미하기에, 개발자가 제어할 수 있는 함수인 generate() 만 테스트 할 수 있게 되면 좋겠네요!

Reply:
이부분은 잘 와닿지않네요..

Reply:
아래처럼 개발자가 제어하길 윈하는 부분이었던 generate() 함수를 아래처럼 테스트해볼 수 있어요.
다른 테스트 코드에서도 제어하기 어려운 랜덤 부분을 개발자가 제어할 수 있게 코드를 작성해보면, 다른 부분들도 쉽게 테스트를 작성하실 수 있게 되어요 :)

```suggestion
        // Arrange
        val randomGeneratorNumber = object : NumberGenerator<Int> {
            override fun generate(): Int {
                return 1
            }
        }
        // Act
        val generatedNumber = randomGeneratorNumber.generate()

        // Assert
        assertThat(generatedNumber).isEqualTo(1)
```

Reply:
감사합니다!
이런걸 말씀하신거군요
테스트코드에서 직접 임시 구현체를 만드는거군요